src/main/scala/omniauth/lib/FacebookProvider.scala

```diff
@@ -56,11 +72,29 @@ class FacebookProvider(val clientId:String, val secret:String) extends OmniauthP
     urlParameters += ("client_secret" -> secret)
     urlParameters += ("code" -> fbCode.toString)
     val tempRequest = :/("graph.facebook.com").secure / "oauth/access_token" <<? urlParameters
-    val accessToken = extractToken(Omniauth.http(tempRequest as_str))
 
-    if(validateToken(accessToken)){
+    implicit val formats = net.liftweb.json.DefaultFormats
+
+    var validToken = false
+    try {
+      val accessTokenStr = Omniauth.http(tempRequest >- { json => json })
+      val accessTokenJSON = parse(accessTokenStr).extract[FacebookTokenResponse]
+      logger.debug("extract token from json " + accessTokenJSON)
+      val accessToken = AuthToken(accessTokenJSON.access_token, None, None, None)
+
+      validToken = validateToken(accessToken)
+
+    } catch {
+      case unknown : Throwable => {
```

Comment:
it's better to use `NonFatal` http://www.scala-lang.org/api/2.9.3/scala/util/control/NonFatal$.html

instead of `Throwable`, because `Throwable` also includes things like out of memory errors which you don't want to really catch